src/Storages/StorageFile.cpp

```diff
@@ -509,6 +497,27 @@ Pipe StorageFile::read(
             else
                 return metadata_snapshot->getColumns();
         };
+        // if we do multiple reads from pipe, we want to check if pipe is a regular file or a pipe
+        if (this_ptr->table_fd_was_used)
+        {
+            if (S_ISREG(this_ptr->table_fd))
+            {
+                if (this_ptr->table_fd_init_offset < 0)
+                {
+                    throw Exception("File descriptor isn't seekable, inside " + this_ptr->getName(), ErrorCodes::CANNOT_SEEK_THROUGH_FILE);
+                }
+                int seekable_table_fd = dup(this_ptr->table_fd);
+                if (lseek(seekable_table_fd, this_ptr->table_fd_init_offset, SEEK_SET) < 0)
+                {
+                    throw Exception("File descriptor isn't seekable, inside " + this_ptr->getName(), ErrorCodes::CANNOT_SEEK_THROUGH_FILE);
+                }
+            }
+            else
+            {
+                // else if fd is a pipeline, then throw an Exception
+                throw Exception("Cannot read from a pipeline twice", ErrorCodes::CANNOT_READ_FROM_FILE_DESCRIPTOR);
```

Comment:
I guess this should mean that this file is not regular.

Reply:
Fixed the comment.